community/kernel/src/test/java/org/neo4j/unsafe/impl/batchimport/input/InputEntityCacherTest.java

```diff
@@ -0,0 +1,282 @@
+    {
+        RecordFormats recordFormats = mock( RecordFormats.class );
+        RecordFormat propertyKeyTokenFormat = getRecordFormatMock( maxPropertyKeyId );
+        RecordFormat labelTokenFormat = getRecordFormatMock( maxLabelId );
+        RecordFormat relationshipTypeTokenFormat = getRecordFormatMock( maxRelationshipTypeId );
+        RecordFormat relationshipGroupTokenFormat = getRecordFormatMock( maxRelationshipGroupId );
+        when( recordFormats.propertyKeyToken() ).thenReturn( propertyKeyTokenFormat );
+        when( recordFormats.labelToken() ).thenReturn( labelTokenFormat );
+        when( recordFormats.relationshipTypeToken() ).thenReturn( relationshipTypeTokenFormat );
+        when( recordFormats.relationshipGroup() ).thenReturn( relationshipGroupTokenFormat );
+        return recordFormats;
+    }
+
+    private RecordFormat getRecordFormatMock( long maxId )
+    {
+        RecordFormat recordFormat = mock( RecordFormat.class );
+        when( recordFormat.getMaxId() ).thenReturn( maxId );
+        return recordFormat;
+    }
+
+    private Randoms getRandoms()
+    {
+        return new Randoms( randomRule.random(), Randoms.DEFAULT );
+    }
+
+    private TestInputEntityCacher getEntityCacher( RecordFormats recordFormats ) throws IOException
+    {
+        return new TestInputEntityCacher( mock( StoreChannel.class ),
+                mock( StoreChannel.class ), recordFormats, 100, 100 );
+    }
+
+    private TestNodeEntityCacher getNodeCacher( RecordFormats recordFormats ) throws IOException
+    {
+        return new TestNodeEntityCacher( mock( StoreChannel.class ),
+                mock( StoreChannel.class ), recordFormats, 100 );
+    }
+
+    private TestRelationshipEntityCacher getRelationshipCacher( RecordFormats recordFormats ) throws IOException
+    {
+        return new TestRelationshipEntityCacher( mock( StoreChannel.class ),
+                mock( StoreChannel.class ), recordFormats, 100 );
+    }
+
+    private class TestInputEntityCacher extends InputEntityCacher
+    {
+        TestInputEntityCacher( StoreChannel channel, StoreChannel header,
+                RecordFormats recordFormats, int bufferSize, int groupSlots ) throws IOException
+        {
+            super( channel, header, recordFormats, bufferSize, groupSlots );
+        }
+    }
+
+    private class TestNodeEntityCacher extends InputNodeCacher
+    {
+        TestNodeEntityCacher( StoreChannel channel, StoreChannel header,
+                RecordFormats recordFormats, int bufferSize ) throws IOException
+        {
+            super( channel, header, recordFormats, bufferSize );
+        }
+    }
+
+    private class TestRelationshipEntityCacher extends InputRelationshipCacher
+    {
+        TestRelationshipEntityCacher( StoreChannel channel, StoreChannel header,
+                RecordFormats recordFormats, int bufferSize ) throws IOException
+        {
+            super( channel, header, recordFormats, bufferSize );
+        }
+    }
```

Comment:
The introduced `Test***EntityCacher` classes make no difference. Can use the extended classes directly.